### ISSUE TITLE
Feature Request: Configure json logging in the docker plugin section.

### ISSUE THREAD

**Issue by powellchristoph** [hashicorp/nomad#6563]
Currently Nomad overwrites the docker daemon logging configuration to `json-file with log rotation (max-file=2 and max-size=2m` and that can be changed within the job's config.

I would like to be able to set the `json-file` configuration options within the docker plugin section on my clients and have it applied to all docker jobs.

**Comment by apollo13** [comment]
Agreed, this would be very useful. @tgross is this something you'd be okay with adding? Ie basically move the defaults into the plugin config, so the user can override it. Or would dropping the default logging config and using whatever dockerd has configured also be an option?

**Comment by tgross** [comment]
> Agreed, this would be very useful. @tgross is this something you'd be okay with adding? Ie basically move the defaults into the plugin config, so the user can override it. Or would dropping the default logging config and using whatever dockerd has configured also be an option?

It seems reasonable on its face; I just took a quick look at where we set those values in [`drivers/docker/driver.go`](https://github.com/hashicorp/nomad/blob/v1.0.4/drivers/docker/driver.go#L865-L877) and it looks like we could have it use the current values as the default, let the plugin config set the values, and let the job override those values. Mostly a matter of getting the HCL schema plumbing for the plugin.

That being said, we're not likely to get to this anytime in the immediate future. I'd be happy to review a PR that takes it on though!
